lib/ui/painting.dart

```diff
@@ -3746,6 +3746,96 @@ class ImageShader extends Shader {
+///
+/// NOTE: This API is in beta and does not yet work on web.
+/// See https://github.com/flutter/flutter/projects/207 for roadmap.
+///
+/// When initializing or updating the [floatUniforms], the length of float
+/// uniforms must match the total number of floats defined as uniforms in
+/// the shader. They will be updated in the order that they are defined.
+///
+/// For example, if there are 3 uniforms: 1 of type float, 1 type float2/vec2,
+/// and 1 of type vec3/float3, and 1 mat2x2 then the length of [floatUniforms]
+/// must be 10.
+///
+/// The uniforms could be updated as follows:
+///
+///   Example fragment shader uniforms.
+///
+///   `uniform float a;`
+///   `uniform vec2 b;`
+///   `uniform vec3 c;`
+///   `uniform mat2x2 d;`
+///
+///   Dart code to update uniforms.
+///
+///   `shader.update(floatUniforms: Float32List.fromList([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]));`
+///
+///   Results of shader uniforms.
+///
+///   a: 1
+///   b: [2, 3]
+///   c: [4, 5, 6]
+///   d: [7, 8, 9, 10] // 2x2 matrix in column-major order
+///
+class FragmentShader extends Shader {
+  @pragma('vm:entry-point')
+
+  // TODO(chriscraws): Add `List<Shader>? children` as a parameter to the
+  // constructor and to [update].
+  // https://github.com/flutter/flutter/issues/85240
+
+  /// Creates a fragment shader from SPIR-V byte data as an input.
+  ///
+  /// The [floatUniforms] can be passed optionally to initialize the
+  /// shader's uniforms. If they are not initially set, they will default
+  /// to 0. They can later be updated by invoking the [update] method.
+  FragmentShader({
+    required ByteBuffer spirv,
+    Float32List? floatUniforms,
+    bool debugPrint = false,
+  }) : super._() {
+    _constructor();
+    final spv.TranspileResult result =
+        spv.transpile(spirv, spv.TargetLanguage.sksl);
+    _uniformFloatCount = result.uniformFloatCount;
+    _init(result.src, debugPrint);
+    update(floatUniforms: floatUniforms ?? Float32List(_uniformFloatCount));
+  }
+
+  late final int _uniformFloatCount;
+
+  void _constructor() native 'FragmentShader_constructor';
+  void _init(String sksl, bool debugPrint) native 'FragmentShader_init';
+
+  /// Updates the uniform values that are supplied to the [FragmentShader]
+  /// and refreshes the shader.
+  ///
+  /// The length of float uniforms must match the total number of
+  /// floats defined as uniforms in the shader. They will be updated
+  /// in the order that they are defined.
+  ///
+  /// See [FragmentShader] for more information on passing uniforms.
+  ///
+  /// This method will aquire additional fields as [FragmentShader] is
+  /// implemented further.
+  void update({
+    Float32List? floatUniforms,
+  }) {
+    if (floatUniforms == null) {
+      throw ArgumentError('No argument was given');
+    }
+    if (floatUniforms.length != _uniformFloatCount) {
+      throw ArgumentError(
+        'FragmentShader floatUniforms size must match given shader');
```

Comment:
here and elsewhere: convention is to end exceptions with a period, iirc.

Reply:
(might be useful to say what the expected length is here, too)

Reply:
added details

Reply:
doesn't seem to have updated here

Reply:
Added actual and expected counts with period